### DISCUSSION TITLE
Why ENGINE_POOL_SIZE_THRESHOLD should be in  [1, 32] ?

### DISCUSSION THREAD

**Discussion by shenbinglife** [apache/kyuubi#2279] [Q&A] [answered]
https://github.com/apache/incubator-kyuubi/blob/6a231519454c5d75aca2120de140c06febf7ae54/kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala#L1027

**Comment by yaooqinn** [comment]
Indeed, it can have an unlimited upper bound.

**Comment by zwangsheng** [comment, marked as answer]
According to #986, this upper limit should be variable.
